packages/core/saved-objects/core-saved-objects-import-export-server-internal/src/import/import_saved_objects.test.ts

```diff
@@ -382,6 +437,214 @@ describe('#importSavedObjectsFromStream', () => {
         };
         expect(mockCreateSavedObjects).toHaveBeenCalledWith(createSavedObjectsParams);
       });
+
+      test('creates managed saved objects', async () => {
+        const options = setupOptions({ createNewCopies: true, managed: true });
+        const collectedObjects = [createObject({ managed: true })];
+        const errors = [createError(), createError()];
+        mockCollectSavedObjects.mockResolvedValue({
+          errors: [errors[0]],
+          collectedObjects,
+          importStateMap: new Map([
+            ['foo', {}],
+            ['bar', { isOnlyReference: true }],
+          ]),
+        });
+        mockCheckReferenceOrigins.mockResolvedValue({
+          importStateMap: new Map([['bar', { isOnlyReference: true, destinationId: 'newId' }]]),
+        });
+        mockValidateReferences.mockResolvedValue([errors[1]]);
+        mockRegenerateIds.mockReturnValue(new Map([['foo', { destinationId: `randomId1` }]]));
+
+        await importSavedObjectsFromStream(options);
+        // assert that the importStateMap is correctly composed of the results from the three modules
+        const importStateMap: ImportStateMap = new Map([
+          ['foo', { destinationId: `randomId1` }],
+          ['bar', { isOnlyReference: true, destinationId: 'newId' }],
+        ]);
+        const createSavedObjectsParams = {
+          objects: collectedObjects,
+          accumulatedErrors: errors,
+          savedObjectsClient,
+          importStateMap,
+          overwrite,
+          namespace,
+          managed: options.managed,
+        };
+        expect(mockCreateSavedObjects).toHaveBeenCalledWith(createSavedObjectsParams);
+      });
+    });
+
+    describe('managed option', () => {
+      test('if not provided, does not override existing property when already present, defaults to false for others', async () => {
```

Comment:
Just looked briefly, but I think this test should focus on asserting the paramaters that we send to the create API
```
expect(mockCreateSavedObjects).toHaveBeenCalledWith(createSavedObjectsParams)
```

We already test `createSavedObjects` function's behaviour in other places, so mocking ES responses and checking the success results are less useful and I think these are already covered in existing tests.

Reply:
We already indirectly test that the params don't include the new option in [`creates saved objects`](https://github.com/elastic/kibana/pull/155677/files/4eeb530cbdaebb95ef2e86f585d5160942368ed0#diff-a9d5649e465d1c21ae59943c888fc5fdb8f0570652898723230f320098f3bc44R285). 

I'll be more explicit in this one then.

Reply:
I refactored all of them in this group of tests to assert on the params rather than the objects getting created.